### ISSUE TITLE
Problem with Pay upon Invoice

### ISSUE THREAD

**Issue by msbskriehs** [woocommerce/woocommerce-paypal-payments#2884]
### Describe the Bug

When the customer chooses to pay via Pay upon Invoice, the order in the backend no longer changes from "On Hold" to "Processing." Additionally, the PayPal transaction ID is not visible in the order details. However, the emails confirming that the purchase on account was completed are being received.

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.4.2
Legacy REST API Package Version: ✔ 1.0.4
Action Scheduler Version: ✔ 3.8.2
Log Directory Writable: ✔
WP Version: 6.7.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: de_DE
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 8.1.30-bookworm.1
PHP Post Max Size: 32 MB
PHP Time Limit: 30
PHP Max Input Vars: 2000
cURL Version: 7.88.1
OpenSSL/3.0.15

SUHOSIN Installed: –
MySQL Version: 8.0.39-30
Max Upload Size: 32 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

amn_mi-lite: 3
amn_smtp: 1
attachment: 9834
blocks: 3
custom_css: 1
dwqa-question: 2
f12_captcha_log: 108
feedback: 87
jp_img_sitemap: 5
jp_img_sitemap_index: 1
jp_sitemap: 1
jp_sitemap_master: 1
jp_vid_sitemap: 1
mailpoet_page: 1
nav_menu_item: 28
oembed_cache: 1
page: 61
popup: 1
popup_theme: 8
popupbuilder: 2
post: 935
product: 255
product_variation: 1675
revision: 328
shop_coupon: 87
shop_order: 11865
shop_order_refund: 206
slide: 7
ux_template: 4
wp_global_styles: 1
wpcf7_contact_form: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (39) ###

Admin Menu Editor: von Janis Elsts – 1.12.4
Angell EYE Updater: von Angell EYE – 2.0.21
Captcha for WordPress: von Forge12 Interactive GmbH – 2.0.682
CleverReach® WP: von CleverReach GmbH & Co. KG – 1.5.20
Complianz | GDPR/CCPA Cookie Consent: von Really Simple Plugins – 7.1.5
Honeypot for Contact Form 7: von WPExperts – 2.1.7
Contact Form 7: von Takayuki Miyoshi – 6.0.1
Google Analytics for WordPress by MonsterInsights: von MonsterInsights – 9.2.2
Google Analytics Opt-Out: von WP-Buddy – 2.3.6
Google Language Translator: von Translate AI Multilingual Solutions – 6.0.20
Google for WooCommerce: von WooCommerce – 2.9.0
OMGF: von Daan von Daan.dev – 5.9.1
Imagify: von Imagify Image Optimizer – Optimize Images & Convert WebP & Avif – 2.2.3.2
Jetpack: von Automattic – 14.0
Kadence Blocks - Gutenberg Blöcke für Page Builder Features: von Kadence WP – 3.3.7
One Stop Shop für WooCommerce: von vendidero – 1.6.2
Plugins Last Updated Column: von Fastmover – 0.1.3
Page Builder by SiteOrigin: von SiteOrigin – 2.31.0
Smart Manager For WooCommerce – Stock Management, Bulk Edit & more...: von StoreApps – 5.31.0
UpdraftPlus – Sichern/Wiederherstellen: von TeamUpdraft
DavidAnderson – 1.24.11

uptain Conversion-Rate Optimierung: von uptain GmbH – 2.4.1
PayPal Ratenzahlung for WooCommerce: von Angell EYE – 1.1.3
PayPal PLUS für WooCommerce: von Inpsyde GmbH – 2.2.1
WooCommerce Brands: von WooThemes – 1.5.0 (Update auf Version 1.7.7 ist verfügbar)
Sofort payment gateway: von Sven Wagener – 1.6.0
Germanized für WooCommerce: von vendidero – 3.18.5
WooCommerce Legacy REST API: von WooCommerce – 1.0.4
WooPayments: von WooCommerce – 8.5.1 (Update auf Version 8.6.0 ist verfügbar)
WooCommerce PayPal Payments: von WooCommerce – 2.9.4
WooCommerce Drucke Rechnungen/Packlisten: von SkyVerge – 3.13.3 (Update auf Version 3.13.7 ist verfügbar)
WooCommerce Sequential Order Numbers Pro: von SkyVerge – 1.21.2 (Update auf Version 1.21.3 ist verfügbar)
WooCommerce Shipping & Tax: von WooCommerce – 2.8.3
WooCommerce Shipment Tracking: von WooThemes – 1.6.0 (Update auf Version 2.5.2 ist verfügbar)
WooCommerce: von Automattic – 9.4.2 (Update auf Version 9.4.3 ist verfügbar)
WP Mail SMTP: von WP Mail SMTP – 4.2.0
WP Rocket: von WP Media – 3.17.3.1
WP Rollback: von WP Rollback – 2.0.7
Custom Product Tabs for WooCommerce: von Code Parrots – 1.8.5
YITH WooCommerce Color, Image & Label Variation Swatches: von YITH – 2.13.0

### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php

### Must Use Plugins (2) ###

Asset CleanUp Pro: Plugin Filtering: von Gabriel Livan – 1.0
uncache-admin.php: von  –

### Settings ###

Legacy API Enabled: ✔
Force SSL: –
Currency: EUR (€)
Currency Position: right_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔
Enforce Approved Product Download Directories: –
HPOS feature enabled: –
Order datastore: WC_Order_Data_Store_CPT
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 10 Tage
Level threshold: –
Log directory size: 19 kB

### WC Pages ###

Shop-Basis: #35214 - /shop/
Warenkorb: #66036 - /warenkorb/ -  Enthält den [woocommerce_cart]-Shortcode
Kasse: #66037 - /kasse/ -  Enthält den [woocommerce_checkout]-Shortcode
Mein Konto: #66038 - /mein-konto/
Allgemeine Geschäftsbedingungen: #35969 - /agb/

### Theme ###

Name: Flatsome Child
Version: 3.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Flatsome
Parent Theme Version: 3.19.7
Parent Theme Author URL: https://uxthemes.com
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php
flatsome/woocommerce/cart/cart-shipping.php
flatsome/woocommerce/cart/cart.php
flatsome/woocommerce/cart/mini-cart.php
flatsome/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/checkout/form-coupon.php
flatsome/woocommerce/checkout/order-received.php
flatsome/woocommerce/checkout/terms.php
flatsome/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product-cat.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-single-product.php
flatsome/woocommerce/content-widget-product.php
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/header.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/rating.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php
flatsome/woocommerce/notices/notice.php
flatsome/woocommerce/notices/success.php
flatsome/woocommerce/product-searchform.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/rating.php
flatsome/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php


### WooPayments ###

Version: 8.5.1
Connected to WPCOM: ❌ Nein
Logging: Deaktiviert

### WooCommerce PayPal Payments ###

Onboarded: ✔
Shop country code: DE
WooCommerce currency supported: ✔
Advanced Card Processing available in country: ✔
Pay Later messaging available in country: ✔
Webhook status: –
PayPal Vault enabled: –
ACDC Vault enabled: –
Logging enabled: –
Reference Transactions: –
Used PayPal Checkout plugin: –
Subscriptions Mode: Disabled
PayPal Shipping Callback: –
Apple Pay: ✔
Google Pay: ✔
Fastlane: –

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: product-data-views
experimental-blocks
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system
blueprint
reactify-classic-payments-settings
use-wp-horizon

Daily Cron: ✔ Next scheduled: 2024-12-07 11:57:13 +01:00
Options: ✔
Notes: 224
Onboarding: completed

### Action Scheduler ###

Abgeschlossen: 2.056
Oldest: 2024-11-05 17:54:51 +0000
Newest: 2024-12-06 17:37:29 +0000

Fehlgeschlagen: 7
Oldest: 2023-01-27 16:39:21 +0000
Newest: 2023-05-30 17:51:08 +0000

In Bearbeitung: 1
Oldest: –
Newest: –

Ausstehend: 31
Oldest: 2024-12-06 17:37:03 +0000
Newest: 2024-12-07 13:11:29 +0000


### Status report information ###

Generated at: 2024-12-06 18:37:57 +01:00

**Comment by SydeKrystian** [comment]
Hello @msbskriehs 

It seems that your GitHub issue is primarily a support request. While we value your input and feedback, GitHub is predominantly used for our development activities. For a prompt and detailed response, we kindly ask that you reach out directly to our [support team](https://woocommerce.com/document/woocommerce-paypal-payments/#get-help).  

When you contact our support team, please include a copy of your [WooComnmerce system report](https://woocommerce.com/document/understanding-the-woocommerce-system-status-report/). This will help us get to the bottom of your issue more quickly. 

We'll be closing this GitHub issue for now. However, once you reach out to our support team, rest assured they'll be ready to assist you with your concerns. Thank you for your understanding!

Kind regards,
Krystian
